### PR TITLE
Teeekay recalculate

### PR DESCRIPTION
**Pull request by teeekay**
Calculates the minimum stopping distance based on not exceeding a maximum jerk value 100 ms after changing direction of acceleration. Now uses a distance in produce_slowdown.  Also cuts back on logging, but moves some state changes to WARN

### CODE REVIEW

**Comment by edufford** [comment]
I couldn't really follow the logic line-by-line so I took a test run and will look at the data instead.  If you are confident that this is an improvement, we can merge it to make testing with the other latest updates easier.  Also, I'm starting to use the TL ground truth pass through logic to do testing by the actual stop lights to see various transitions, so is it possible to disable the `dyn_test_stoplight_wp` waypoint by default?  By the current settings, I can only move the waypoint around, but can't disable it.

**Comment by edufford** [comment]
I realized that when the TL ground truth pass-through is actively sending TL waypoints, it already overwrites the `dyn_test_stoplight_wp` manually set waypoint.  Could you modify the `dyn_test_stoplight_wp` logic to allow a range of -1 to 11010, where if it is set > -1 then force overwrite to keep `next_tl_wp = dyn_test_stoplight_wp`?

**Comment by teeekay** [comment]
Added changes for `dyn_test_stoplight_wp` as requested

File: ros/src/waypoint_updater/waypnt_updater.py [hunk 1]
```diff
@@ -473,49 +626,62 @@ def set_waypoints_velocity(self):
...
                 if self.state != 'slowdown':
-                    rospy.loginfo("now start to slowdown")
+                    rospy.logwarn("now start to slowdown")
                     self.state = 'slowdown'
-                recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
+                if dist_to_tl - self.dyn_tl_buffer > self.min_stop_distance:
+                    recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
                                                self.lookahead_wps,
-                                               accel_ratio)
+                                               dist_to_tl - self.dyn_tl_buffer)
```
Comment (edufford) [inline ros/src/waypoint_updater/waypnt_updater.py]: If the dist_to_tl activates within the normal (self.stopping_distance + buffer) and is also already closer than the self.min_stop_distance, this means that the car can't stop in time and will just keep going?  If so, can you add an else condition here that at least makes a log notification of this so we can confirm this behavior during testing?

File: ros/src/waypoint_updater/waypnt_updater.py [hunk 2]
```diff
@@ -473,49 +626,62 @@ def set_waypoints_velocity(self):
...
-                self.waypoints[self.final_waypoints_start_ptr].get_v(),
-                0.0, self.default_accel/accel_ratio)
-
-        rospy.loginfo("dist_to_tl = {:4.3f}, stopping_dist = {:4.3f}, state = {}"
-                      .format(dist_to_tl, stopping_distance, self.state))
+            if (math.fabs(self.prev_step_v -
+                    self.waypoints[self.final_waypoints_start_ptr].get_v()) > 0.1
+                    or 
+                    math.fabs(self.prev_step_a -
+                    self.waypoints[self.final_waypoints_start_ptr].get_a()) > 0.1):
```
Comment (edufford) [inline ros/src/waypoint_updater/waypnt_updater.py]: Is this condition just to reduce unnecessary recalculations of the stopping distances until reaching the next waypoint where the speed or accel is changing, especially for the maintain speed case? If so, could you add a comment to explain this for reference?

File: ros/src/waypoint_updater/waypnt_updater.py [hunk 3]
```diff
@@ -473,49 +626,62 @@ def set_waypoints_velocity(self):
         if self.waypoints[self.final_waypoints_start_ptr].get_v() == 0.0:
             # we are stopped
             offset = 0
-            stopping_distance = 0.0
             if self.state != 'stopped':
-                rospy.loginfo("We have stopped.")
+                rospy.logwarn("We have stopped.")
                 self.state = 'stopped'
+                self.stopping_distance = 0.0
+                self.min_stop_distance = 0.01
```
Comment (edufford) [inline ros/src/waypoint_updater/waypnt_updater.py]: Why is self.min_stop_distance re-initialized to 0.01 instead of 0.0?

**Pull request merged**